core/server/common/src/main/java/alluxio/cli/validation/ClusterConfConsistencyValidationTask.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.cli.validation;
+
+import alluxio.Configuration;
+import alluxio.PropertyKey;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+
+/**
+ * Task for validating system limit for current user.
+ */
+public final class ClusterConfConsistencyValidationTask extends AbstractValidationTask {
+  private static final Set<String> PROPERTIES_TO_IGNORE =  new HashSet<>(Arrays.asList(
+      PropertyKey.USER_HOSTNAME.getName(),
+      PropertyKey.WORKER_HOSTNAME.getName(),
+      PropertyKey.MASTER_HOSTNAME.getName()
```

Comment:
Should the cluster share the same master hostname?

Reply:
Also, I think capacity-related configurations like `alluxio.worker.memory.size` can be ignored? It's for managing local resources.

Reply:
> Should the cluster share the same master hostname?

In HA mode each master is supposed to have their own host name set on alluxio.master.hostname

> Also, I think capacity-related configurations like alluxio.worker.memory.size can be ignored? It's for managing local resources.

I was debating whether we should ignore those properties or not. I am leaning towards keeping it given this task only marks it as a warning. Most users configure Alluxio with same storage settings across entire cluster, it can still be useful to warn user that some nodes are having different storage settings in case they forget to sync the configurations.

Reply:
sgtm. Thanks.

Reply:
I'm concerned that this will generate so many false positives that they will drown out real misconfigurations. For each property, there are 3 possibilities:
1. The property must be the same across the servers it applies to. If we find that it isn't, the check should show an error.
2. The property should almost always be the same across the servers it applies to. If we find that it isn't, we should show a warning.
3. The property can reasonably be different across the servers it applies to.

I suggest recording which of the three categories each PropertyKey falls under when we construct the PropertyKey in `PropertyKey.java`.

It's also important to take into consideration that most configuration applies only to masters or only to workers. We shouldn't complain that a master node doesn't contain a worker-only property, or vice versa.